dot/types/body.go

```diff
@@ -174,3 +127,19 @@ func (b *Body) HasExtrinsic(target Extrinsic) (bool, error) {
 
 	return false, nil
 }
+
+// AsEncodedExtrinsics decodes the body into an array of SCALE encoded extrinsics
+func (b *Body) AsEncodedExtrinsics() ([]Extrinsic, error) {
+	decodedExts := *b
+	ret := make([]Extrinsic, len(decodedExts))
+	var err error
+
+	for i, ext := range decodedExts {
+		ret[i], err = scale.Marshal(ext)
```

Comment:
The marshaling part is correct, but I think this function should be better named as `AsEncodedBytesArray`. That would avoid the confusion that it is causing. And the value returned by this function is being converted into `[][]byte` and `[]string` anyway, so returning `[][]byte` is better

Reply:
Are we sure that the marshalling here is correct? On my branch jimmy/integrateScaleIntoTypes I had my own tests (https://github.com/ChainSafe/gossamer/blob/0cb69f5f2946a36c3fb7982102af274f2d2b3e50/dot/types/block_test.go#L111) for block encoding written which I compare the encoding to the expected byte string (hard coded so I don't test against itself) and when I merge these changes that test fails because the encoding doesn't equal what is expected. So there appears to be difference between the encodings, but I am not certain what the correct format is. For example, the body.AsEncodedExtrinsics() function specifies that it is scale encoded but that is not actually the case because there is not length prefix for the encoded slice. I am not positive if it should be scale encoded or if its just incorrect documentation but it does appear that the body encoding now is different than it was before. Perhaps @noot could help provide some clarification on this?